### PR TITLE
Attempting to fix issue #845

### PR DESCRIPTION
**Pull request by jerrylususu**
Changes:
1. Add `errors` and `stackTraces` to Help.ColorScheme
2. Change the `internalHandleParseException` method to use `colorScheme.errorText(ex.getMessage())`, instead of just `ex.getMessage()`
3. Add a method called `exceptionToColorString` to convert exceptions to ANSI colored strings.
4. Change the `handleUnhandled` method to use `exceptionToColorString`

Tests: existing tests updated, no new test introduced
Documentation: JavaDoc updated, no change to other documentations

### CODE REVIEW

File: src/main/java/picocli/CommandLine.java [hunk 1]
```diff
@@ -1929,10 +1929,30 @@ public int execute(String... args) {
...
 
+    /**
+     * Convert an {@code Exception} to a {@code String} , with message and stack traces extracted and colored
+     * according to {@code ColorScheme}.
+     * @param ex the {@code Exception} to be converted
+     * @param colorScheme the {@code ColorScheme} to use
+     * @return converted and colored {@code String}
+     */
+    private static String exceptionToColorString(Exception ex, Help.ColorScheme existingColorScheme){
+        Help.ColorScheme colorScheme = new Help.ColorScheme.Builder(existingColorScheme).applySystemProperties().build();
```
Comment (remkop) [inline src/main/java/picocli/CommandLine.java:1945]: Out of interest: why not use the specified `existingColorScheme` directly? Is there a reason to make a copy?

File: src/main/java/picocli/CommandLine.java [hunk 2]
```diff
@@ -1929,10 +1929,30 @@ public int execute(String... args) {
         }
     }
     private static int handleUnhandled(Exception ex, CommandLine cmd, int defaultExitCode) {
-        ex.printStackTrace(cmd.getErr());
+        cmd.getErr().print(exceptionToColorString(ex, cmd.getColorScheme()));
+        cmd.getErr().flush();
```
Comment (remkop) [inline src/main/java/picocli/CommandLine.java:1933]: Is this call to `flush()` necessary?

File: src/main/java/picocli/CommandLine.java (hunk 1 above)
Comment (jerrylususu) [reply src/main/java/picocli/CommandLine.java:1945]: Sorry... Just noticed this after a few days. One reason I'm using a copy here is not sure if the system property about the color setting might be changing during runtime, and another reason is this is how `Help` constructor works, so I'd like to comply with what's existing here...

File: src/main/java/picocli/CommandLine.java (hunk 2 above)
Comment (jerrylususu) [reply src/main/java/picocli/CommandLine.java:1933]: I added this line to fix a bug I find during testing. It seems that Java's `PrintWriter` won't flush until a line separator is met, so I'd like to add one `flush()` here to make sure the message is indeed printed. In retrospect, this line might not be necessary, as error messages generated with `exceptionToColorString` should already have line separators in it. But... just to be safe?

**Pull request merged**